File: 1044-find-common-characters/1044-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        first=words[0]
        ans=[]
        for i in first:
            if i not in ans:
                val=first.count(i)
                for j in range(1,len(words)):
                    val=min(val,words[j].count(i))
                    if val==0:
                        break
                for k in range(val):
                    ans.append(i)
                # print(ans)
        return ans
            


        

File: 1044-find-common-characters/1044-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
       vector<string> frequency;
       for (auto &chr:words[0])
       {
            int count=0;
            for(int i=1;i<words.size();i++)
            {
                // cout<<words[i];
                for(auto &ch:words[i])
                {
                    if (ch==chr)
                    {
                        count+=1;
                        ch='-';
                        break;
                    }
                }
            }
            // cout<<count<<endl;
            if (count+1==words.size())
            {
                count=0;
                frequency.push_back(std::string(1,chr));
            }
       } 
       return frequency;
    }
};